src/types/page.rs

```diff
@@ -169,9 +264,36 @@ impl PageRangesPart {
     }
 }
 
-impl From<u64> for PageRangesPart {
-    fn from(value: u64) -> Self {
-        Self::SinglePage((value as u32).into())
+impl From<i32> for PageRangesPart {
+    fn from(value: i32) -> Self {
+        Self::SinglePage(value.into())
+    }
+}
+
+impl TryFrom<u32> for PageRangesPart {
+    type Error = TryFromIntError;
+
+    fn try_from(value: u32) -> Result<Self, Self::Error> {
+        let value: i32 = value.try_into()?;
+        Ok(Self::SinglePage(value.into()))
+    }
+}
+
+impl TryFrom<i64> for PageRangesPart {
+    type Error = TryFromIntError;
+
+    fn try_from(value: i64) -> Result<Self, Self::Error> {
+        let value: i32 = value.try_into()?;
+        Ok(Self::SinglePage(value.into()))
+    }
+}
+
+impl TryFrom<u64> for PageRangesPart {
+    type Error = TryFromIntError;
+
+    fn try_from(value: u64) -> Result<Self, Self::Error> {
+        let value: i32 = value.try_into()?;
+        Ok(Self::SinglePage(value.into()))
```

Comment:
Removing `From<u64>` (which, at a glance, appears to be the correct decision) caused a compilation error in `csl-json` when resolving page variables, as `PageRanges::From(u64)` no longer works:

https://github.com/typst/hayagriva/blob/a1a27dc6ac58ddde3b4deed3ab441619d9e2b25a/src/csl/taxonomy.rs#L807-L815

Is doing a fallible conversion and returning `None` on error an appropriate choice here? Not sure if that's how `Option` is supposed to be interpreted in this particular internal API. We could also just saturate specifically here and call it a day, though I'd be inclined to indicate failure somehow.